crates/bindings-csharp/BSATN.Codegen/Utils.cs

```diff
@@ -1,30 +1,33 @@
 namespace SpacetimeDB.Codegen;
 
-using System.Collections;
-using System.Collections.Immutable;
 using System.Text;
 using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
-using static System.Collections.StructuralComparisons;
 
 public static class Utils
 {
-    // Even `ImmutableArray<T>` is not deeply equatable, which makes it a common
-    // pain point for source generators as they must use only cacheable types.
+    // None of the standard collections - even `ImmutableArray<T>` - are deeply equatable,
+    // which makes it a common pain point for source generators as they must use only cacheable types.
     // As a result, everyone builds their own `EquatableArray<T>` type.
-    public readonly record struct EquatableArray<T>(ImmutableArray<T> Array) : IEnumerable<T>
+    // In our case, we use a `List<T>`-backed type because we want to extend it dynamically in a few places.
```

Comment:
Is there any way to prevent extending a list that's already stored as a cache key?

Reply:
We could split up types into separate "builder" and end list. Or use `ImmutableList` which has a more costly implementation, but, well, it's immutable. I'm not sure if it's worth the effort, but if you think so, I could add something.